### ISSUE TITLE
BitLocker-opencl new ASAN error 

### ISSUE THREAD

**Issue by claudioandre** [magnumripper/JohnTheRipper#3040]
### IMPORTANT
- Related to something new in code

```
Testing: BitLocker-opencl, BitLocker [SHA256 AES OpenCL]... =================================================================
==13721== ERROR: AddressSanitizer: stack-buffer-overflow on address 0x7ffcc67bb107 at pc 0x904e3b bp 0x7ffcc67bb080 sp 0x7ffcc67bb078
WRITE of size 1 at 0x7ffcc67bb107 thread T0
    #0 0x904e3a in set_key /home/travis/build/claudioandre/JohnTheRipper/src/opencl_bitlocker_fmt_plug.c:674
    #1 0x841d64 in fmt_self_test_body /home/travis/build/claudioandre/JohnTheRipper/src/formats.c:845
    #2 0x843e81 in fmt_self_test /home/travis/build/claudioandre/JohnTheRipper/src/formats.c:1684
    #3 0x822941 in benchmark_format /home/travis/build/claudioandre/JohnTheRipper/src/bench.c:367
    #4 0x8246b2 in benchmark_all /home/travis/build/claudioandre/JohnTheRipper/src/bench.c:816
    #5 0x852583 in john_run /home/travis/build/claudioandre/JohnTheRipper/src/john.c:1617
    #6 0x853c3c in main /home/travis/build/claudioandre/JohnTheRipper/src/john.c:2000
    #7 0x7f2691feef44 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x21f44)
    #8 0x4067f2 in _start (/home/travis/build/claudioandre/JohnTheRipper/run/john+0x4067f2)
Address 0x7ffcc67bb107 is located at offset 87 in frame <set_key> of T0's stack:
  This frame has 2 object(s):
    [32, 87) 'tmp'
    [128, 183) 'tmp2'
HINT: this may be a false positive if your program uses some custom stack unwind mechanism or swapcontext
      (longjmp and C++ exceptions *are* supported)
SUMMARY: AddressSanitizer: stack-buffer-overflow /home/travis/build/claudioandre/JohnTheRipper/src/opencl_bitlocker_fmt_plug.c:674 set_key
Shadow bytes around the buggy address:
  0x100018cef5d0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100018cef5e0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100018cef5f0: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100018cef600: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100018cef610: 00 00 00 00 00 00 f1 f1 f1 f1 00 00 00 00 00 00
=>0x100018cef620:[07]f4 f2 f2 f2 f2 00 00 00 00 00 00 07 f4 00 00
  0x100018cef630: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100018cef640: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100018cef650: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100018cef660: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
  0x100018cef670: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
Shadow byte legend (one shadow byte represents 8 application bytes):
  Addressable:           00
  Partially addressable: 01 02 03 04 05 06 07 
  Heap left redzone:     fa
  Heap righ redzone:     fb
  Freed Heap region:     fd
  Stack left redzone:    f1
  Stack mid redzone:     f2
  Stack right redzone:   f3
  Stack partial redzone: f4
  Stack after return:    f5
  Stack use after scope: f8
  Global redzone:        f9
  Global init order:     f6
  Poisoned by user:      f7
  ASan internal:         fe
==13721== ABORTING
```

**Comment by kholia** [comment]
@e-ago Can you please look into this? Thanks!

**Comment by magnumripper** [comment]
Did this not happen with the the bot's ASan builds in the PR?

**Comment by magnumripper** [comment]
> Did this not happen with the the bot's ASan builds in the PR?

That was #3025. No bots had problems with it.

**Comment by magnumripper** [comment]
Apparently I fixed this in 745723abb5 without even knowing there was an issue for it.
